core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -1337,7 +1339,7 @@ public static PlanAndMappings coerce(PlanBuilder subPlan, List<Expression> expre
                         subPlan.getRoot(),
                         assignments.build()));
 
-        return new PlanAndMappings(subPlan, mappings.build());
+        return new PlanAndMappings(subPlan, mappings);
```

Comment:
`ImmutableMap.copyOf`

(`PlanAndMappings` constructor should _also_ do that, but currently it doesn't)

Reply:
That's actually unnecessary here. This method doesn't hold on to the map, so it doesn't care what `PlanAndMappings` does with it. `PlanAndMappings` should be updated to do it, though.

Reply:
> That's actually unnecessary here. This method doesn't hold on to the map

yes, but we don't want downstream code to become dependent on the map being mutable

(not important, nvrm)